file/src/main/scala/akka/stream/alpakka/file/model.scala

```diff
@@ -21,72 +21,127 @@ final class TarArchiveMetadata private (
     val filePathPrefix: Option[String],
     val filePathName: String,
     val size: Long,
-    val lastModification: Instant
+    val lastModification: Instant,
+    val linkIndicator: Byte
 ) {
   val filePath = filePathPrefix match {
     case None => filePathName
     case Some(prefix) => prefix + "/" + filePathName
   }
 
+  def isDirectory: Boolean =
+    linkIndicator == TarArchiveMetadata.linkIndicatorDirectory || filePathName.endsWith("/")
+
   override def equals(obj: Any): Boolean = {
     obj match {
       case that: TarArchiveMetadata =>
         this.filePathPrefix == that.filePathPrefix &&
         this.filePathName == that.filePathName &&
         this.size == that.size &&
-        this.lastModification == that.lastModification
+        this.lastModification == that.lastModification &&
+        this.linkIndicator == that.linkIndicator
       case _ => false
     }
   }
 
-  override def hashCode(): Int = Objects.hash(filePathPrefix, filePathName, Long.box(size), lastModification)
+  override def hashCode(): Int =
+    Objects.hash(filePathPrefix, filePathName, Long.box(size), lastModification, Byte.box(linkIndicator))
 
   override def toString: String =
     "TarArchiveMetadata(" +
     s"filePathPrefix=$filePathPrefix," +
     s"filePathName=$filePathName," +
     s"size=$size," +
-    s"lastModification=${lastModification})"
+    s"lastModification=$lastModification," +
+    s"linkIndicator=${linkIndicator.toChar})"
 }
 
 object TarArchiveMetadata {
+
+  /**
+   * Constants for the `linkIndicator` flag.
+   */
+  val linkIndicatorNormal: Byte = '0'
```

Comment:
Small enum(-ish) for that?

Reply:
There exist custom extensions to these values. For now, we just care about normal and directory.